File: src/main/java/dev/kensa/context/TestContextHolder.java
package dev.kensa.context;

public final class TestContextHolder {

    private static final ThreadLocal<TestContext> holder = new ThreadLocal<>();

    public static void bindToThread(TestContext testContext) {
        holder.set(testContext);
    }

    public static TestContext testContext() {
        return holder.get();
    }

    public static void clearFromThread() {
        holder.remove();
    }
}

File: src/main/java/dev/kensa/context/TestContextHolder.kt
package dev.kensa.context

object TestContextHolder {
    private val holder = ThreadLocal<TestContext>()

    @JvmStatic
    fun bindToThread(testContext: TestContext) {
        holder.set(testContext)
    }

    @JvmStatic
    fun testContext(): TestContext = holder.get()

    @JvmStatic
    fun clearFromThread() {
        holder.remove()
    }
}